sdk-http-vertx/src/main/java/dev/restate/sdk/http/vertx/RequestHttpServerHandler.java

```diff
@@ -152,22 +177,82 @@ private Executor currentContextExecutor(Context currentContext) {
+      // Compute response and write it back
+      DeploymentManifestSchema response = this.restateEndpoint.handleDiscoveryRequest();
+
+      Buffer responseBuffer;
+      try {
+        responseBuffer = Buffer.buffer(MANIFEST_OBJECT_MAPPER.writeValueAsBytes(response));
+      } catch (JsonProcessingException e) {
+        LOG.warn("Error when writing out the manifest POJO", e);
+        request.response().setStatusCode(INTERNAL_SERVER_ERROR.code()).end();
+        return;
+      }
+
+      request
+          .response()
+          .setStatusCode(OK.code())
+          .putHeader(X_RESTATE_SERVER_KEY, X_RESTATE_SERVER_VALUE)
+          .putHeader(
+              CONTENT_TYPE,
+              serviceDiscoveryProtocolVersionToContentType(serviceDiscoveryProtocolVersion))
+          .end(responseBuffer);
+    } else {
+      request
+          .response()
+          .setStatusCode(UNSUPPORTED_MEDIA_TYPE.code())
+          .setStatusMessage(
+              "Service endpoint does not support the accepted service discovery protocol versions.")
+          .end();
+    }
+  }
+
+  private static Discovery.ServiceDiscoveryProtocolVersion
+      selectSupportedServiceDiscoveryProtocolVersion(String accept) {
+    // assume V1 in case nothing was set
+    if (accept == null || accept.isEmpty()) {
+      return Discovery.ServiceDiscoveryProtocolVersion.V1;
+    }
+
+    final String[] supportedVersions = accept.split(",");
+
+    Discovery.ServiceDiscoveryProtocolVersion maxVersion =
+        Discovery.ServiceDiscoveryProtocolVersion.SERVICE_DISCOVERY_PROTOCOL_VERSION_UNSPECIFIED;
+
+    for (String versionString : supportedVersions) {
+      final Optional<Discovery.ServiceDiscoveryProtocolVersion> optionalVersion =
+          parseServiceDiscoveryProtocolVersion(versionString.trim());
+
+      if (optionalVersion.isPresent()) {
+        final Discovery.ServiceDiscoveryProtocolVersion version = optionalVersion.get();
+        if (version.getNumber() > maxVersion.getNumber()) {
+          maxVersion = version;
+        }
+      }
     }
 
-    request
-        .response()
-        .setStatusCode(OK.code())
-        .putHeader(X_RESTATE_SERVER_KEY, X_RESTATE_SERVER_VALUE)
-        .putHeader(CONTENT_TYPE, APPLICATION_JSON)
-        .end(responseBuffer);
+    return maxVersion;
+  }
+
+  private static Optional<Discovery.ServiceDiscoveryProtocolVersion>
+      parseServiceDiscoveryProtocolVersion(String versionString) {
+    if (versionString.equals("application/vnd.restate.endpointmanifest.v1+json")) {
+      return Optional.of(Discovery.ServiceDiscoveryProtocolVersion.V1);
+    }
+    return Optional.empty();
+  }
+
+  private static String serviceDiscoveryProtocolVersionToContentType(
+      Discovery.ServiceDiscoveryProtocolVersion version) {
+    if (Objects.requireNonNull(version) == Discovery.ServiceDiscoveryProtocolVersion.V1) {
+      return "application/vnd.restate.endpointmanifest.v1+json";
+    }
+    return "";
```

Comment:
Throw a new illegal status exception here.

Reply:
Will change it.